apps/obsidian/src/components/RelationshipSettings.tsx

```diff
@@ -0,0 +1,296 @@
+import { useState } from "react";
+import {
+  DiscourseRelation,
+  DiscourseNode,
+  DiscourseRelationType,
+} from "../types";
+import { Notice } from "obsidian";
+import { usePlugin } from "./PluginContext";
+
+const RelationshipSettings = () => {
+  const plugin = usePlugin();
+  const [discourseRelations, setDiscourseRelations] = useState<
+    DiscourseRelation[]
+  >(() => plugin.settings.discourseRelations ?? []);
+  const [hasUnsavedChanges, setHasUnsavedChanges] = useState(false);
+
+  const findNodeById = (id: string): DiscourseNode | undefined => {
+    return plugin.settings.nodeTypes.find((node) => node.id === id);
+  };
+
+  const findRelationTypeById = (
+    id: string,
+  ): DiscourseRelationType | undefined => {
+    return plugin.settings.relationTypes.find((relType) => relType.id === id);
+  };
+
+  const handleRelationChange = async (
+    index: number,
+    field: keyof DiscourseRelation,
+    value: string,
+  ): Promise<void> => {
+    const updatedRelations = [...discourseRelations];
+
+    if (!updatedRelations[index]) {
+      updatedRelations[index] = {
+        sourceId: "",
+        destinationId: "",
+        relationshipTypeId: "",
+      };
+    }
+
+    updatedRelations[index][field] = value;
+    setDiscourseRelations(updatedRelations);
+    setHasUnsavedChanges(true);
+  };
+
+  const handleAddRelation = (): void => {
+    const updatedRelations = [
+      ...discourseRelations,
+      {
+        sourceId: "",
+        destinationId: "",
+        relationshipTypeId: "",
+      },
+    ];
+    setDiscourseRelations(updatedRelations);
+    setHasUnsavedChanges(true);
+  };
+
+  const handleDeleteRelation = async (index: number): Promise<void> => {
+    const updatedRelations = discourseRelations.filter((_, i) => i !== index);
+    setDiscourseRelations(updatedRelations);
+    plugin.settings.discourseRelations = updatedRelations;
+    await plugin.saveSettings();
+    new Notice("Relation deleted");
+  };
+
+  const handleSave = async (): Promise<void> => {
+    for (const relation of discourseRelations) {
+      if (
+        !relation.relationshipTypeId ||
+        !relation.sourceId ||
+        !relation.destinationId
+      ) {
+        new Notice("All fields are required for relations.");
+        return;
+      }
+    }
+
+    const relationKeys = discourseRelations.map(
+      (r) => `${r.relationshipTypeId}-${r.sourceId}-${r.destinationId}`,
+    );
+    if (new Set(relationKeys).size !== relationKeys.length) {
+      new Notice("Duplicate relations are not allowed.");
+      return;
+    }
+
+    plugin.settings.discourseRelations = discourseRelations;
+    await plugin.saveSettings();
+    new Notice("Relations saved");
+    setHasUnsavedChanges(false);
+  };
+
+  return (
+    <div className="discourse-relations">
+      <h3>Node Type Relations</h3>
+
+      {plugin.settings.nodeTypes.length === 0 && (
```

Comment:
Let's combine these with a ternary so we don't get both.
image.png

Reply:
![image](https://github.com/user-attachments/assets/e1119915-a606-4c0e-9cb7-7aa711528679)